### ISSUE TITLE
Restart required if scripts directory is empty

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#1229]
If the `.retriever/scripts` directory is empty when starting a Python session the retriever reports that there are no scripts even after successfully running `check_for_updates`.

```python
In [1]: import retriever                                                                                                                                                                                           

In [2]: retriever.dataset_names()                                                                                                                                                                                  
Out[2]: []

In [3]: retriever.check_for_updates()                                                                                                                                                                              
Downloading scripts: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 97/97 [00:33<00:00,  2.99files/s]

In [4]: retriever.dataset_names()                                                                                                                                                                                  
Out[4]: []
```

If you restart Python then everything works as expected.

See https://github.com/ropensci/rdataretriever/issues/173 for initial report.

**Comment by ethanwhite** [comment]
This results from 398ca50484d290c59eecf0d64b9e1d5c4bc0c7ce. In general this commit is great - we don't want to constantly be reloading scripts, but we do need to be able to after we've checked for updates. I played with this a bit, but I found the current structure of the code related to `global_script_list`, `StoredScripts`, and `SCRIPT_LIST` to be a bit confusing and couldn't figure out how to call for a reloading of the scripts from `check_for_updates`.

@zhangcandrew - do you have any thoughts on how to do this?

**Comment by henrykironde** [comment]
I also came across that https://github.com/henrykironde/retriever/issues/117
I think Andrew will be able to figure that out.

**Comment by ethanwhite** [comment]
:+1: We should fix this and roll a new release before we release the new version of `rdataretriever` because this issue will happen for all first time users on R. It's confusing enough that I think it's worth fixing before we generate new users through advertising around the release.

**Comment by zhangcandrew** [comment]
I think this should be fixed in #1231 

I agree, the structure is a bit confusing. We ran into the problem because reload_scripts was being called by the singleton on instantiation before check_for_updates was called later on in the code. As a result, the global singleton scripts were set to nothing after a retriever reset. 

I've made it now so that upon instantiation of the SCRIPT_LIST it will do a check_for_updates if necessary. I think this should work.
